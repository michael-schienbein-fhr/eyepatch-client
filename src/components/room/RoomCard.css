.RoomCard {
 display: flex;
 justify-content: center;
 background-color: black;
  border: 1px solid;
  border-color: #6c757d;
  border-radius: .25rem;
}

.room-card-body{
  display: flex;
  justify-content: start;
  text-decoration: none;
  width: 100%;
  
}

.RoomCard-link{
  width: 100%;
  text-decoration: none;
}

.delete{
  margin-right: 10px;
  color: whitesmoke;
  text-shadow: -1px 0 #6c757d, 0 1px #6c757d, 1px 0 #6c757d, 0 -1px #6c757d;
  font-size: 1.5rem;
  font-weight: bold;
}
