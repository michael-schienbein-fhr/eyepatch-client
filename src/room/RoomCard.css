.RoomCard {
 display: flex;
 justify-content: center;
 /* width: 100%;  */
}

.RoomCard-link{
  text-decoration: none;
  width: 100%;
  background-color: black;
  border: 1px solid;
  border-color: #6c757d;
  border-radius: .25rem;
}