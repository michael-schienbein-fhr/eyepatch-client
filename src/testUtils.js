import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const demoQueueVideo = {
  username: 'testuser2',
  type: 'video',
  action: 'add',
  text: '"testVideoTitle" added to queue for user: "testuser2".',
  videoId: 'testVideoId',
  title: 'testVideoTitle',
  description: 'testDescription',
  thumbnail: 'testThumbnail'
}
const demoSearchVideo = { "kind": "youtube#searchResult", "etag": "3_5Vn-wWRzIN5QC7aOEWvZsmMwI", "id": { "kind": "youtube#video", "videoId": "GjbibR6DV1s" }, "snippet": { "publishedAt": "2015-08-11T17:03:57Z", "channelId": "UCi_ZH3bYHidyHsETH5hbTFQ", "title": "Dusky - Skin Deep (Official Video)", "description": "Dusky's EP 'Ordinary World' is out now. iTunes:http://po.st/DuskyowEP Spotify: http://po.st/DuskyOWSp Beatport: http://po.st/OrdinaryWorldEP \"This EP was all ...", "thumbnails": { "default": { "url": "https://i.ytimg.com/vi/GjbibR6DV1s/default.jpg", "width": 120, "height": 90 }, "medium": { "url": "https://i.ytimg.com/vi/GjbibR6DV1s/mqdefault.jpg", "width": 320, "height": 180 }, "high": { "url": "https://i.ytimg.com/vi/GjbibR6DV1s/hqdefault.jpg", "width": 480, "height": 360 } }, "channelTitle": "DuskyMusic", "liveBroadcastContent": "none", "publishTime": "2015-08-11T17:03:57Z" } }

const testRoom = {
  "id": 1,
  "roomOwner": "testadmin",
  "roomName": "testadmin",
  "roomMembers": null,
  "videoQueue": null,
  "hasPass": true,
  "created_at": "2021-10-22T10:34:59.443Z"
}
const UserProvider =
  ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
  );

export { UserProvider, demoUser, demoQueueVideo, demoSearchVideo, testRoom };

