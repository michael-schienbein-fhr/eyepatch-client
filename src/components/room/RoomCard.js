import './RoomCard.css';
import { useContext, useState } from "react";
import { Link } from "react-router-dom";
import UserContext from "../../auth/UserContext";

function RoomCard({ id, roomName, hasPass, roomOwner, deleteRoom }) {
  const { currentRoom, currentUser } = useContext(UserContext);
  // const [isActive, setIsActive] = useState(false);
  // const toggleActive = () => {
  //   setIsActive(true);
  // }
  const deleteButton = (currentUser.username === roomOwner || currentUser.isAdmin === true) ? <div className="delete" onClick={() => deleteRoom(id)}>X</div> : null

  return (
    <div className="RoomCard col-md-6 col-lg-4 mx-0 px-1 py-1">
      {hasPass === true ?
        (
          <div className="room-card-body">
            <Link className="RoomCard-link"
              to={currentRoom && currentRoom.id === id
                ? `/rooms/private/${currentRoom.id}`
                : `/rooms/private/${id}/login`}>
              <h5 className="card-title">
                Room Name: {roomName}
              </h5>
              <p><small>Room Id: {id}</small></p>
              <p><small>Password Protected: ✔️</small></p>
            </Link>
          </div>
        )
        :
        (<div className="room-card-body">
          <Link className="RoomCard-link" to={`/rooms/${id}`}>
            <h5 className="card-title">
              Room Name: {roomName}
            </h5>
            <p><small>Room Id: {id}</small></p>
            <p><small>Password Protected: ❌</small></p>
          </Link>
        </div>
        )}
      {deleteButton}

    </div>
  );
}

export default RoomCard;

