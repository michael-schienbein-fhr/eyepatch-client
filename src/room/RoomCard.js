import './RoomCard.css';
import React, { useContext } from "react";
import { Link } from "react-router-dom";
import UserContext from "../auth/UserContext";
import EyepatchApi from '../api/api';
function RoomCard({ id, roomName, hasPass, roomOwner }) {
  const { currentRoom, currentUser } = useContext(UserContext);

  const deleteRoom = async function (){
    let room = await EyepatchApi.deleteRoom(id);
    console.log(id);
  }
  return (
    <div className="RoomCard col-md-6 col-lg-4 mx-0 px-1 py-1">
      {currentUser.username === roomOwner ? <div onClick={deleteRoom}>X</div> : null}
      {hasPass === true ?
        (<Link className="RoomCard-link"
          to={currentRoom && currentRoom.id === id
            ? `/rooms/private/${currentRoom.id}`
            : `/rooms/private/${id}/login`}>
          <div className="card-body">
            <h5 className="card-title">
              Room Name: {roomName}
            </h5>
            <p><small>Room Id: {id}</small></p>
            <p><small>Password Protected: ✔️</small></p>
          </div>
        </Link>)
        :
        (<Link className="RoomCard-link" to={`/rooms/${id}`}>

          <div className="card-body">
            <h5 className="card-title">
              Room Name: {roomName}
            </h5>
            <p><small>Room Id: {id}</small></p>
            <p><small>Password Protected: ❌</small></p>
          </div>
        </Link>
        )}
    </div>
  );
}

export default RoomCard;

