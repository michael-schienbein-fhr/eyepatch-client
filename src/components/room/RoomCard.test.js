import React from "react";
import { render } from "@testing-library/react";
import RoomCard from "./RoomCard";
import { UserProvider, testRoom } from "../../testUtils";
import { MemoryRouter, Route } from "react-router";

it("matches snapshot ", function () {

  const { asFragment } = render(
    <MemoryRouter >
      <UserProvider>
        <Route>
          <RoomCard />
        </Route>
      </UserProvider>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});