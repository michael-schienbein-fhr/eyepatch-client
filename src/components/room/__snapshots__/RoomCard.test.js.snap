// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot  1`] = `
<DocumentFragment>
  <div
    class="RoomCard col-md-6 col-lg-4 mx-0 px-1 py-1"
  >
    <div
      class="room-card-body"
    >
      <a
        class="RoomCard-link"
        href="/rooms/undefined"
      >
        <h5
          class="card-title"
        >
          Room Name: 
        </h5>
        <p>
          <small>
            Room Id: 
          </small>
        </p>
        <p>
          <small>
            Password Protected: ‚ùå
          </small>
        </p>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
